File: src/favoringIsolatedMutability/java/create/HollywoodActor.java
/***
 * Excerpted from "Programming Concurrency on the JVM",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/vspcon for more book information.
***/
package com.agiledeveloper.pcj;

import akka.actor.UntypedActor;

@SuppressWarnings("unchecked")
public class HollywoodActor extends UntypedActor {
  public void onReceive(final Object role) {
    System.out.println("Playing " + role + 
      " from Thread " + Thread.currentThread().getName());
  }  
}

File: src/concurrency-code/favoringIsolatedMutability/scala/create/HollywoodActor.scala
package com.agiledeveloper.pcj

import akka.actor.Actor

//START:CODE
class HollywoodActor extends Actor {
  def receive = {
    case role => 
    println("Playing " + role + 
      " from Thread " + Thread.currentThread().getName())
  }  
}
//END:CODE
